include/ttmlir/Dialect/TTNN/IR/TTNNOps.td

```diff
@@ -638,6 +638,29 @@ def TTNN_MaxPool2dOp : TTNN_NamedDPSOp<"max_pool2d"> {
     let hasVerifier = 1;
 }
 
+def TTNN_ClampOp : TTNN_Op<"clamp"> {
+    let summary = "Clamp op.";
+    let description = [{
+      Clamp tensor values to a specified range.
+
+      Example:
+        min: 2.000000+00
+        input: [[0, 1, 2, 3, 4, 5, 6, 7]]
+        max: 5.000000+00
+
+        "ttnn.clamp"(%arg0) <{max = 2.000000e+00 : f32, min = 5.000000e+00 : f32}>
+        -> %out = [[2, 2, 2, 3, 4, 5, 5, 5]]
+    }];
+
+    let arguments = (ins Variadic<AnyRankedTensor>:$inputs,
```

Comment:
Why do we have inputs defined as variadic? Shouldn't we use a single input? Like in TTNN API:
https://github.com/tenstorrent/tt-metal/blob/a5d9979eb38ca89d7dea9ccaf5c3c684a7f6b2d8/ttnn/cpp/ttnn/operations/eltwise/unary/unary_composite.hpp#L119

Reply:
We are using `TTNNToFlatbuffer.cpp::createEltwiseOp(...)` to handle both `binary` and `unary` ops. Variadic inputs provide a way to handle all eltwise ops with same code. I can refactor the code and have separate function to handle unary ops. Then we won't need variadic inputs. I can refactor it in a separate PR. what do you suggest?

Reply:
Honestly, I'm not sure, but I think we are breaking some logical structure here. If we leave it like this, we are not making an abstraction of unary op in MLIR, but we do in the runtime. We should either inherit from the unary operation table definition in MLIR or handle this operation separately in runtime, similar to how we manage the Softmax operation. What are your thoughts?

Reply:
I am more inclined towards refactoring `createEltwiseOp` into 1) `createEltwiseUnaryOp` and 2) `createEltwiseBinaryOp` (and match it with runtime abstraction). We should also convert variadic input/output definitions to non-variadic definitions for unary/binary ops which will make TTIR/TTNN dialect more precise. 
My initial runtime implementation handled clamp op independently similar to Softmax op but we decided to handle clamp op as unary to match tt-metal implementation.

Reply:
@mmanzoorTT @sdjordjevicTT chiming in here about the runtime implementation. Softmax is implemented separately because it's not modelled as an eltwise op in ttnn. It's a separate normalization op, therefore we model it as an independent normalization op in runtime. Clamp however is modelled as an eltwise unary op in ttnn so we should model it as a unary op in our runtime as well. This follows the general north star of our runtime reflecting the modelling of ttnn runtime as closely as possible

Reply:
Thanks for the clarification folks, really appreciated it. I see, in general for unary ops, we are extending the TTNN_ElementwiseUnaryOp which further extends TTNN_ElementwiseOp and it actually has variadic inputs and outputs and is modeled as DPS.

I'm wondering if we could implement a similar structure for elementwise operations that are not DPS. Instead of simply inheriting from `TTNN_Op`, could we establish a hierarchy as follows:

TTNN_ClampOp -> TTNN_ElementwiseUnaryOpNonDPS -> TTNN_ElementwiseOpNonDPS -> TTNN_Op

Reply:
Thanks @jnie-TT for the input. I agree with you @sdjordjevicTT. I'll create an issue for this change and handle them in a separate PR. We have few other non DPS ops and I prefer to change all of them together.

Reply:
Great, thanks!

Reply:
issue created https://github.com/tenstorrent/tt-mlir/issues/1270. Please have a look @sdjordjevicTT